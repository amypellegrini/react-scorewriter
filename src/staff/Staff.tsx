import React from "react";

import "./Staff.css";

/**
 * We're extending Staff with SVGAttributes to keep DRY
 */
export interface StaffProps extends React.SVGAttributes<HTMLOrSVGElement> {
}

export default function Staff( {
  x = 0,
  y = 0,
  viewBox = `0 0 100 100`,
  height = "100",
  width = "100",
  preserveAspectRatio = "none"
}: StaffProps ) {
  return (
    <svg
      data-testid="staff"
      x={x}
      y={y}
      viewBox={viewBox}
      width={width}
      height={height}
      preserveAspectRatio={preserveAspectRatio}
    >
      <line
        x1="0"
        y1="1"
        x2="100"
        y2="1"
        strokeWidth="1"
        stroke="black"
      />
      <line
        x1="0"
        y1="25"
        x2="100"
        y2="25"
        strokeWidth="1"
        stroke="black"
      />
      <line
        x1="0"
        y1="50"
        x2="100"
        y2="50"
        strokeWidth="1"
        stroke="black"
      />
      <line
        x1="0"
        y1="75"
        x2="100"
        y2="75"
        strokeWidth="1"
        stroke="black" />
      <line
        x1="0"
        y1="99"
        x2="100"
        y2="99"
        strokeWidth="1"
        stroke="black"
      />
    </svg>
  );
}
