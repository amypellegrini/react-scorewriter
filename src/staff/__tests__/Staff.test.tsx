import React from "react";
import { render } from "@testing-library/react";
import Staff from "../Staff";

describe( "Staff", () => {
  it( "Renders with defaults", () => {
    const { container: received } = render( <Staff /> );
    expect( received ).toMatchSnapshot();
  } );
  it( "Renders with sizes", () => {
    const { container: received } = render(
      <div style={{
        width: "100%",
        height: 100
      }}>
        <Staff
          width={"100%"}
          height={100}
          preserveAspectRatio="xMidYMid meet"
        />
      </div>
    );
    expect( received ).toMatchSnapshot();
  } );
} );
