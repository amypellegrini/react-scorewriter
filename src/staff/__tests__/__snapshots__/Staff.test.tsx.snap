// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Staff Renders with defaults 1`] = `
<div>
  <svg
    data-testid="staff"
    height="100"
    preserveAspectRatio="none"
    viewBox="0 0 100 100"
    width="100"
    x="0"
    y="0"
  >
    <line
      stroke="black"
      stroke-width="1"
      x1="0"
      x2="100"
      y1="1"
      y2="1"
    />
    <line
      stroke="black"
      stroke-width="1"
      x1="0"
      x2="100"
      y1="25"
      y2="25"
    />
    <line
      stroke="black"
      stroke-width="1"
      x1="0"
      x2="100"
      y1="50"
      y2="50"
    />
    <line
      stroke="black"
      stroke-width="1"
      x1="0"
      x2="100"
      y1="75"
      y2="75"
    />
    <line
      stroke="black"
      stroke-width="1"
      x1="0"
      x2="100"
      y1="99"
      y2="99"
    />
  </svg>
</div>
`;

exports[`Staff Renders with sizes 1`] = `
<div>
  <div
    style="width: 100%; height: 100px;"
  >
    <svg
      data-testid="staff"
      height="100"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 100 100"
      width="100%"
      x="0"
      y="0"
    >
      <line
        stroke="black"
        stroke-width="1"
        x1="0"
        x2="100"
        y1="1"
        y2="1"
      />
      <line
        stroke="black"
        stroke-width="1"
        x1="0"
        x2="100"
        y1="25"
        y2="25"
      />
      <line
        stroke="black"
        stroke-width="1"
        x1="0"
        x2="100"
        y1="50"
        y2="50"
      />
      <line
        stroke="black"
        stroke-width="1"
        x1="0"
        x2="100"
        y1="75"
        y2="75"
      />
      <line
        stroke="black"
        stroke-width="1"
        x1="0"
        x2="100"
        y1="99"
        y2="99"
      />
    </svg>
  </div>
</div>
`;
